src/resolve/connectionResolver.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection, Logger } from '@salesforce/core';
+import { RegistryAccess, registry as defaultRegistry, MetadataType } from '../registry';
+import { standardValueSet } from '../registry/standardvalueset';
+import { FileProperties, StdValueSetRecord, ListMetadataQuery } from '../client/types';
+import { normalizeToArray } from '../utils';
+import { MetadataComponent } from './types';
+export interface ResolveConnectionResult {
+  components: MetadataComponent[];
+}
+
+/**
+ * Resolve MetadataComponents from an org connection
+ */
+export class ConnectionResolver {
+  protected logger: Logger;
+  private connection: Connection;
+  private registry: RegistryAccess;
+
+  public constructor(connection: Connection, registry = new RegistryAccess()) {
+    this.connection = connection;
+    this.registry = registry;
+    this.logger = Logger.childFromRoot(this.constructor.name);
+  }
+
+  public async resolve(excludeManagedComponents = true): Promise<ResolveConnectionResult> {
+    const Aggregator: Array<Partial<FileProperties>> = [];
+    const childrenPromises: Array<Promise<FileProperties[]>> = [];
+    const componentTypes: Set<MetadataType> = new Set();
+
+    const componentPromises: Array<Promise<FileProperties[]>> = [];
+    for (const type of Object.values(defaultRegistry.types)) {
+      componentPromises.push(this.listMembers({ type: type.name }));
+    }
+    for await (const componentResult of componentPromises) {
+      Aggregator.push(...componentResult);
+      for (const component of componentResult) {
+        const componentType = this.registry.getTypeByName(component.type);
+        componentTypes.add(componentType);
+        const folderContentType = componentType.folderContentType;
+        if (folderContentType) {
+          childrenPromises.push(
+            this.listMembers({
+              type: this.registry.getTypeByName(folderContentType).name,
+              folder: component.fullName,
+            })
+          );
+        }
+      }
+    }
+
+    for (const componentType of componentTypes) {
+      const childTypes = componentType.children?.types;
+      if (childTypes) {
+        Object.values(childTypes).map((childType) => {
+          childrenPromises.push(this.listMembers({ type: childType.name }));
+        });
+      }
+    }
+
+    for await (const childrenResult of childrenPromises) {
+      Aggregator.push(...childrenResult);
+    }
+
+    const components = Aggregator.filter(
+      (component) =>
+        !(excludeManagedComponents && component.namespacePrefix && component.manageableState !== 'unmanaged')
```

Comment:
@mshanemc @shetzel do you think this is the correct way to exclude "managed components" or should we exclude all components with a "namespacePrefix"? Is `installedEditable` a `manageableState` we should handle here as well? Thanks!

Reply:
I honestly don't know.  Is this something that should be exposed in `FromConnectionOptions`?  I'm not sure what each field means to know how it would affect the filter.

Reply:
@shetzel I've added a `componentFilter` to `FromConnectionOptions` to cover possible changes.